openerp/addons/base/ir/ir_translation.py

```diff
@@ -304,8 +304,25 @@ def _get_ids(self, cr, uid, name, tt, lang, ids):
                 translations[res_id] = value
         return translations
 
+    CACHED_MODELS = {'ir.model.fields', 'ir.ui.view'}
+
+    def _modified_model(self, model_name):
+        """ Invalidate the ormcache if necessary, depending on ``model_name``.
+        This should be called when modifying translations of type 'model'.
+        """
+        if model_name in self.CACHED_MODELS:
+            self.clear_caches()
+
+    def _modified(self):
```

Comment:
It is not used with the old API, so it is not strictly necessary.
But it's better with it than without it.